ext/drwrap/drwrap.c

```diff
@@ -2300,8 +2361,9 @@ drwrap_event_bb_analysis(void *drcontext, void *tag, instrlist_t *bb, bool for_t
 }
 
 static dr_emit_flags_t
-drwrap_event_bb_insert(void *drcontext, void *tag, instrlist_t *bb, instr_t *inst,
-                       bool for_trace, bool translating, void *user_data)
+drwrap_event_bb_insert_where(void *drcontext, void *tag, instrlist_t *bb, instr_t *inst,
```

Comment:
I don't understand why we need to support separate `inst` and `where`. As "wrapping" implies adding instrumentation just-before and just-after, wouldn't it be wrong to pass different values for inst (the instructionto be wrapped) and where (where instrumentation has to be placed)?

I see that drbbdup would pass different values for `inst` and `where` in this case:
https://github.com/DynamoRIO/dynamorio/blob/4d8c8372681367051c091ca5659780a396ac1c4f/ext/drbbdup/drbbdup.c#L1272

Is this the use-case? Worth adding a comment to explain it.

Reply:
> As "wrapping" implies adding instrumentation just-before and just-after, wouldn't it be wrong to pass different values for inst (the instructionto be wrapped) and where (where instrumentation has to be placed)?

There are several cases these days where the instruction to monitor and where to put the monitoring are different.  One is emulation, and for that things like the drmemtrace tracer code already have some separation of the two (e.g., in instrument_memref()).  Another is drbbdup, which can't clone the block-final instr if it's a branch or syscall -- so it has just one copy after all the duplicated sequences, but it has to put the monitoring code up inside each duplicated case sequence.

Reply:
I see. Worth documenting this in a comment.

Reply:
Done.